### PR TITLE
Adding user management

### PR DESCRIPTION
**Pull request by koggs**
### What we have learned
- Generating user model using devise gem

### CODE REVIEW

File: app/views/recipes/index.html.haml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+%h1 FoodHub
+= link_to 'Sign up',  new_user_registration_path
```
Comment (tochman) [inline app/views/recipes/index.html.haml]: You probably want this link to be visible on all pages throughout the application. Why not place it in the main layout template?

File: db plane [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  ingredients
+  steps
+  likes
+  belongs_to :user
+  has_many :comments, :as => :commentable
+
+Comment
+  belongs_to :commentable, :polymorphic => true
+  belongs_to :recipe
+  belongs_to :user
```
Comment (tochman) [inline db plane]: What sort of file is this? Is it Markdown, Text? Please add a suffix

File: features/step_definitions/assertion_steps.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+Given("I visit the site") do
+  visit root_path
+end
```
Comment (tochman) [inline features/step_definitions/assertion_steps.rb]: This is NOT an assertion

File: features/user_creates_account.feature [hunk 4]
```diff
@@ -0,0 +1,14 @@
+Feature: User creates an account
+  As a user
+  In order to use the site
+  I have to sign up and create an account
+
+  Scenario: User can sign up
+    Given I visit the site
+    When I click "Sign up"
+    # And I fill in "Name" with "Maran"
```
Comment (tochman) [inline features/user_creates_account.feature]: Please remove commented out code

File: spec/models/user_spec.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (tochman) [inline spec/models/user_spec.rb]: Please do not skip the unit test part of the Acceptance - Unit Test Cycle

**Review by tochman — CHANGES_REQUESTED** [review]

**Comment by tochman** [comment]
Please note the error on Semaphore `Don't know how to build task 'ci:tests' (see --tasks)` Please make sure you have the latest code from `upstream` on your feature branch. I can see that the rake task (`lib/tasks/ci.rake`) is missing on your branch.

File: features/user_creates_account.feature (hunk 4 above)
Comment (koggs) [reply features/user_creates_account.feature]: We want to add name field to the sign up using devise.

Comment (tochman) [reply features/user_creates_account.feature]: Is it part of this PR? Adding custom attributes is a bit tricky and should be extracted to a separate feature/story imo.

File: app/views/layouts/application.html.haml [hunk 6]
```diff
@@ -7,4 +7,7 @@
     = stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track': 'reload'
     = javascript_include_tag 'application', 'data-turbolinks-track': 'reload'
   %body
+    %p.notice= notice
+    %p.alert= alert
```
Comment (tochman) [inline app/views/layouts/application.html.haml:11]: What does these 2 lines do for you?

**Review by tochman — COMMENTED** [review]
Please see comment on layout file

File: app/views/layouts/application.html.haml (hunk 6 above)
Comment (koggs) [reply app/views/layouts/application.html.haml:11]: these are to show the messages like "successfully logged in" etc...
they come with devise

Comment (tochman) [reply app/views/layouts/application.html.haml:11]: I’ll let it pass, but I would prefer a conditional here.

**Pull request merged**